src/components/Facility/ConsultationDetails/ConsultationFeedTab.tsx

```diff
@@ -50,7 +51,23 @@ export const ConsultationFeedTab = (props: ConsultationTabProps) => {
     }
   }, []);
 
-  const asset = preset?.asset_bed.asset_object;
+  const bedObjectId = props.consultationData.current_bed?.bed_object?.id || "";
+  const bedsQuery = useQuery(routes.listAssetBeds, {
+    query: {
+      bed_object: bedObjectId,
+      limit: 50,
+    },
+  });
+
+  const bedLocationId = bed?.location_object?.id;
+
+  const matchingAsset = useMemo(() => {
+    return bedsQuery.data?.results.find(
+      (bedItem) => bedItem.asset_object?.location_object?.id === bedLocationId,
+    )?.asset_object;
+  }, [bedsQuery.data, bedLocationId]);
```

Comment:
if bedsQuery (better known as assetBedsQuery) is already filtered by a bed, why filter by location? We do not allow linking an asset with a bed with different locations anyhow in the first place.

```suggestion
```

Reply:
@rithviknishad 
Now i found that assetBedsQuery is not filtering by bed. 

```javascript
const filteredAssets = useMemo(() => {
  return assetBedsQuery.data?.results.filter(
    (bedItem) =>
      bedItem.bed_object?.id === bedId &&
      bedItem.asset_object?.asset_class === "ONVIF"
  );
}, [assetBedsQuery.data, bedId]);
```
  
 i need to filter the fetched values by bedId and asset_class.
 However, I have a question: if a bed has multiple cameras, should I display all camera assets?